src/python/pants/backend/python/util_rules/lockfile.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""
+Helpers for tools to generate their lockfiles
+TODO:
+- TODO: helpful error messages on Tools missing required fields
```

Comment:
Could be nice with a ticket (with ref to issue number) here..

Reply:
https://github.com/pantsbuild/pants/issues/18022